With `nix build .#pkgsCross.aarch64-multiplatform.naja`, I'm unable to cross-compile this from x86_64-linux to aarch64:

```log
 last 25 log lines:
 > [ 42%] Compiling Cap'n Proto schema snl_implementation.capnp
 > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash: line 1: /nix/store/5pbqr4hhyphm1fmfv56y6279vig684xd-capnproto-aarch64-unknown-linux-gnu-1.0.2/bin/capnp: cannot execute binary file: Exec format error
 > make[2]: *** [src/snl/snl/serialization/capnp/CMakeFiles/naja_snl_dump.dir/build.make:81: src/snl/snl/serialization/capnp/snl_interface.capnp.c++] Error 126
 > make[2]: *** Waiting for unfinished jobs....
 > [ 43%] Building CXX object src/snl/formats/verilog/CMakeFiles/naja_snl_verilog.dir/backend/SNLVRLDumper.cpp.o
 > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash: line 1: /nix/store/5pbqr4hhyphm1fmfv56y6279vig684xd-capnproto-aarch64-unknown-linux-gnu-1.0.2/bin/capnp: cannot execute binary file: Exec format error
 > make[2]: *** [src/snl/snl/serialization/capnp/CMakeFiles/naja_snl_dump.dir/build.make:74: src/snl/snl/serialization/capnp/snl_common.capnp.c++] Error 126
 > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash: line 1: /nix/store/5pbqr4hhyphm1fmfv56y6279vig684xd-capnproto-aarch64-unknown-linux-gnu-1.0.2/bin/capnp: cannot execute binary file: Exec format error
 > make[2]: *** [src/snl/snl/serialization/capnp/CMakeFiles/naja_snl_dump.dir/build.make:88: src/snl/snl/serialization/capnp/snl_implementation.capnp.c++] Error 126
 > make[1]: *** [CMakeFiles/Makefile2:864: src/snl/snl/serialization/capnp/CMakeFiles/naja_snl_dump.dir/all] Error 2
 > make[1]: *** Waiting for unfinished jobs....
 > [ 45%] Building CXX object src/snl/snippets/cpp/CMakeFiles/snl_snippet.dir/SNLSnippet.cpp.o
 > [ 46%] Building CXX object src/dnl/CMakeFiles/naja_dnl.dir/DNL.cpp.o
 > [ 46%] Building CXX object src/snl/formats/liberty/CMakeFiles/naja_snl_liberty.dir/SNLBooleanTree.cpp.o
 > [ 46%] Linking CXX executable snl_snippet
 > [ 46%] Built target snl_snippet
 > [ 46%] Linking CXX static library libnaja_snl_liberty.a
 > [ 46%] Built target naja_snl_liberty
 > [ 46%] Linking CXX shared library libnaja_dnl.so
 > [ 46%] Built target naja_dnl
 > [ 46%] Linking CXX static library libnaja_snl_verilog.a
 > [ 46%] Built target naja_snl_verilog
 > [ 47%] Linking CXX static library libnaja_snl_visual.a
 > [ 47%] Built target naja_snl_visual
 > make: *** [Makefile:136: all] Error 2
```

> Fixed. Though I don't really understand why "libnaja_bne.so" wasn't put in the lib output automatically.

Yeah, I'm quite curious as well.

--- a/pkgs/by-name/na/naja/package.nix
+++ b/pkgs/by-name/na/naja/package.nix
@@ -12,16 +12,17 @@
   pkg-config,
   python3,
   tbb_2021_11,
+  buildPackages,
 }:
 stdenv.mkDerivation {
   pname = "naja";
-  version = "0-unstable-2024-08-27";
+  version = "0-unstable-2025-01-13";
 
   src = fetchFromGitHub {
     owner = "najaeda";
     repo = "naja";
-    rev = "ca7a544d16abb31d6992e702ccbd97be3a644c08";
-    hash = "sha256-lmgXv2nmmjKph0Tf9ZvV3kQBtbiGXYA7jrE77cgM+KU=";
+    rev = "ffc29daa22e02565b2a0a108f8e65236cdee413a";
+    hash = "sha256-XGlgSUHSpHxNrms50pOQ9eoGZ6y79Rbm/sDYW2C4qsg=";
     fetchSubmodules = true;
   };
 
@@ -31,17 +32,32 @@ stdenv.mkDerivation {
     "dev"
   ];
 
+  # disable building tests for cross build
+  postPatch =
+    lib.optionalString (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      substituteInPlace CMakeLists.txt \
+        --replace-fail 'enable_testing()' "" \
+        --replace-fail 'add_subdirectory(test)' ""
+      substituteInPlace thirdparty/yosys-liberty/CMakeLists.txt \
+        --replace-fail 'add_subdirectory(test)' ""
+    ''
+    + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      patchShebangs --build test/test_utils/diff_files.py
+    '';
+
   strictDeps = true;
 
-  nativeBuildInputs = [
-    bison
-    capnproto
-    cmake
-    doxygen
-    flex
-    pkg-config
-    python3
-  ];
+  nativeBuildInputs =
+    [
+      bison
+      cmake
+      doxygen
+      flex
+      pkg-config
+    ]
+    ++ lib.optionals (stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+      python3 # test scripts
+    ];
 
   buildInputs = [
     boost
@@ -49,14 +65,24 @@ stdenv.mkDerivation {
     flex # include dir
     libdwarf-lite
     tbb_2021_11
+    python3
   ];
 
   cmakeFlags = [
     (lib.cmakeBool "CPPTRACE_USE_EXTERNAL_LIBDWARF" true)
     (lib.cmakeBool "CPPTRACE_USE_EXTERNAL_ZSTD" true)
+    # provide correct executables for cross
+    (lib.cmakeFeature "Python3_EXECUTABLE" (lib.getExe python3.pythonOnBuildForHost))
+    # TODO: remove these once capnp cross is fixed properly
+    (lib.cmakeFeature "CAPNP_EXECUTABLE" (lib.getExe' buildPackages.capnproto "capnp"))
+    (lib.cmakeFeature "CAPNPC_CXX_EXECUTABLE" (lib.getExe' buildPackages.capnproto "capnpc-c++"))
   ];
 
   doCheck = true;
+
+  postInstall = ''
+    moveToOutput lib/libnaja_bne.so $lib
+  '';
 
   meta = {
     description = "Structural Netlist API (and more) for EDA post synthesis flow development";

